torch/_subclasses/functional_tensor.py

```diff
@@ -249,15 +256,35 @@ def __torch_dispatch__(self, func, types, args=(), kwargs=None):
             )
             return NotImplemented
 
+        def _can_decompose(func):
+            # TODO (tmanlaibaatar)
+            # Eventually, we don't want to decompose any aten op at all
+            # except for aten ops that lie about their schemas (ops
+            # claimed to be functional but actually aren't).
+            # Since we never tested functionalization for every CompositeImplicitAutograd
+            # ops having a direct functional kernel. As a result, we can't safely
+            # turn off decomp for predispatch right now. Therefore, we do best
+            # effort by not decomposing ops that are functional in PreDispatch functionalization
+            # for now.
+            if self._dispatch_key is not None:
```

Comment:
nit: maybe make this `self._dispatch_key == torch._C._DispatchKey.PreDispatch` for clarity (it's kinda annoying that `_dispatch_key` seems like it can be an arbitrary dispatc key here, but we know it's either `None` or `PreDispatch`. Maybe we should just make it a bool)